### PR TITLE
feat(packageserver): support label queries and copy CSV labels onto the package

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**

Copies the labels from the head csv in the default channel onto the labels of the packagemanifest. It also adds default os/arch labels if they are not present.

**Motivation for the change:**

This can be used for filtering packagemanifests; specifically we will use this to filter on an arch label to only display operators that support the architectures supported by the cluster.

This will need to wait for agreement on the label conventions in https://github.com/operator-framework/olm-book/pull/29

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/package-server/provider/registry.go [hunk 1]
```diff
@@ -464,6 +464,12 @@ func newPackageManifest(ctx context.Context, logger *logrus.Entry, pkg *api.Pack
 			defaultElided = defaultElided || pkgChannel.Name == manifest.Status.DefaultChannel
 			continue
 		}
+		manifestLabels := manifest.GetLabels()
+		for k, v := range csv.GetLabels() {
+			manifestLabels[k] = v
+		}
+		setDefaultOsArchLabels(manifestLabels)
+		manifest.SetLabels(manifestLabels)
```
[the remark this answers is not in the document]
Comment (ecordell) [reply pkg/package-server/provider/registry.go]: Good catch - better to be safe here. Updated.

**Comment by njhale** [comment]
Reminder: We should open a followup to fix the empty CSV JSON issue introduced by the new bundle image reference behavior of `operator-registry`.

/lgtm

**Comment by ecordell** [comment]
/retest

some of the tests seem to not be able to clean up after themselves fast enough, might be an underlying issue.

**Pull request merged**